c/extras/ftrl.cc

```diff
@@ -366,12 +344,148 @@ inline double Ftrl::signum(double x) {
 /*
 *  Hash `double` to `uint64_t` based on its bit representation.
 */
-inline uint64_t Ftrl::hash_double(double x) {
+inline uint64_t FtrlModel::hash_double(double x) {
   uint64_t* h = reinterpret_cast<uint64_t*>(&x);
   return *h;
 }
 
 
-void DatatableModule::init_methods_ftrl() {
-  ADDFN(py::ftrl);
+/*
+*  Get a shallow copy of an FTRL model if available.
+*/
+DataTable* FtrlModel::get_model() {
+  if (dt_model != nullptr) {
+    return dt_model->copy();
+  } else {
+    return nullptr;
+  }
+}
+
+
+/*
+*  Set an FTRL model, assuming all the validation is done in `py_ftrl.cc`
+*/
+void FtrlModel::set_model(DataTable* dt_model_in) {
+  dt_model = dtptr(dt_model_in);
```

Comment:
I suspect this may crash... You're taking `dt_model_in`, which is a pointer to a `DataTable*` owned from some python object, and put it inside a `unique_ptr`. At some point this `DataTable` will be freed twice.

Reply:
Yes, you're right. I actually already observed this crash :) Will fix it.